lib/extensions/descendant_loader.rb

```diff
@@ -178,9 +178,15 @@ def cache
 
     def save_cache!
       return unless @cache_dirty
-      cache_path.parent.mkpath
-      cache_path.open('w') do |f|
-        YAML.dump(cache, f)
+
+      # Don't write sti_loader.yml in production, this shouldn't change from what is in the RPM
+      if Rails.env.production?
+        warn "\nSTI cache is out of date in production, check that source files haven't been modified"
```

Comment:
I think we are expecting these to be out of date for a few classes.
Will sharing the name of at least one stale class help us here?

We could set cache_dirty with the name of the first class that made it dirty?

Reply:
We aren't expecting any to be out of date - the ones that Adam found he fixed in the other PR.

Reply:
But I agree with you that maybe the warning should include any known dirty classes.

Reply:
I figured it was simple enough to run `rpm -V manageiq-core` to list what files had been changed, not against listing out the classes in the cache though

Reply:
Previously this would have been _everything_ since the root directory was wrong on everything, but not it should hopefully be only files that were hand-edited